flexget/plugins/sites/lostfilm.py

```diff
@@ -47,151 +38,124 @@ class LostFilm(object):
-                            item['title'],
-                            quality)
-                    else:
-                        log.debug('Item doesn\'t have a title')
-                        continue
-                new_entry = Entry()
-                new_entry['url'] = torrent_link
-                new_entry['title'] = new_title.strip()
-                if series_name_rus:
+
+                # TODO We cannot determine available qualities right now since they are listed on the download page.
+                # So, assume all possible qualities are available and emit entries for them
+                for quality_code, quality_params in QUALITY_MAP.items():
+                    quality, file_ext = quality_params
+                    new_entry = Entry()
+                    new_entry['url'] = "{}?{}".format(V_SEARCH_URL, urllib.urlencode({'c': lostfilm_num, 's': season_num, 'e': episode_num}))
+                    new_entry['name'] = series_name_eng
+                    new_entry['title'] = '{:s} S{:02d}E{:02d} - {:s} [{:s}, {:s}]'.format(series_name_eng, season_num, episode_num, episode_name_eng, quality, file_ext)
+                    new_entry['season'] = season_num
+                    new_entry['episode'] = episode_num
+                    new_entry['quality'] = quality
+                    new_entry['lf_quality_code'] = quality_code
                     new_entry['series_name_rus'] = series_name_rus
-                if episode_name_rus:
                     new_entry['episode_name_rus'] = episode_name_rus
-                if series_name_eng:
                     new_entry['series_name_eng'] = series_name_eng
-                if episode_name_eng:
                     new_entry['episode_name_eng'] = episode_name_eng
-                entries.append(new_entry)
-
+                    entries.append(new_entry)
         return entries
 
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        return entry['url'].startswith(V_SEARCH_URL)
+
+    def url_rewrite(self, task, entry):
+        release_url = entry['url']
+
+        try:
+            response = task.requests.get(release_url)
+        except RequestException as e:
+            log.error('Could not connect to redirect url: %s', e)
+            # TODO raise UrlRewritingError here and below ?
+            return
+
+        page = get_soup(response.content)
+        try:
+            redirect_url = page.head.meta['content'].split('url=')[1]  # retre.org
+        except:
+            log.error('Missing redirect')
+            return
+
+        try:
+            response = task.requests.get(redirect_url)
+        except RequestException as e:
+            log.error('Could not connect to RETRE redirect url: %s', e)
+            return
+
+        page = get_soup(response.content)
+
+        for item in page.findAll('div', 'inner-box--item'):
+            # TODO Cancel/reject entry if the requested quality is not available
```

Comment:
How can I reject an entry on the _urlrewriter_ phase if the requested quality is not available?